### ISSUE TITLE
Hitbox focus on components

### ISSUE THREAD

**Issue by geospatialem** [Esri/calcite-design-system#4633]
### Summary

Complete an audit of all the components to reduce block elements in focus. For instance, the color picker selector.

This came up as an action as part of https://github.com/Esri/calcite-components/issues/1359.

### Actual Behavior

Some components elements are depicted as block elements when focused.

### Expected Behavior

Focus highlights the extent of the element.

#### Components
- [x] `color-picker`: #7378 (verified in `1.5.0-next.30`)
- [x] `link`: #7390 (verified in `1.5.0-next.30`)
- [x] `pagination`: no change
- [x]  `rating`: Focus outline should appear when clicked #7341 
- [x]  `stepper`: no change
- [x] `tabs`: no change
- [x] `switch`: #7390 (verified in `1.5.0-next.30`)

### Reproduction Sample

https://developers.arcgis.com/calcite-design-system/components/color-picker/#sample

### Reproduction Steps

1. Open the [sample page](https://developers.arcgis.com/calcite-design-system/components/color-picker/#sample).
2. Navigate into the color picker using keyboard keys (`tab`) to see the focus highlighting a box for the color picker selector.

### Reproduction Version

beta.82

### Working W3C Example/Tutorial

N/A

### Relevant Info

N/A

**Comment by geospatialem** [comment]
## Audit results:

### 1. `color-picker`
Includes boxy focus on the following elements of the component (displayed in one image below for clarity):
- color handle
- swatch handle and
- RGB and HSV groupings (depicted in a 1px black rectangle)
![color picker screenshot with multiple focus combinations](https://user-images.githubusercontent.com/5023024/209837622-2d635769-b1db-4b63-b209-3ec81ca0236b.png)

### 2. `link`
The hitbox needs a bit more padding, it's a bit hard to discern on the left/right, includes spacing where links could extend multiple lines.
![link focus screenshot simple](https://user-images.githubusercontent.com/5023024/209840220-aa1a90f6-01d9-4471-ac7a-5e353ae56378.png)
![link focus screenshot multiple lines](https://user-images.githubusercontent.com/5023024/209839813-9183da78-2efe-44cd-b458-2e6b072cf2fc.png)

### 3. `pagination` cc https://github.com/Esri/calcite-components/issues/5113
The focus and selection conflict, maybe a differing focus, such as a circle would help differentiate?
![pagination focus screenshot](https://user-images.githubusercontent.com/5023024/209840842-1244de72-7a9a-49f1-8366-c06f0e42dc6f.png)

### 4. `rating`
There is a boxed focus so noting as part of the audit, but it seems to be the best look for the component.
![rating focus screenshot](https://user-images.githubusercontent.com/5023024/209842032-50506ebe-8e8b-4305-a596-eb2ad8d9e6f6.png)

### 5. `stepper` and `stepper-item`
There is a boxed focus so noting as part of the audit, but it seems like a low priority to mitigate, if at all.
![stepper focus screenshot](https://user-images.githubusercontent.com/5023024/209842495-5c61b61e-62e4-4c02-be4f-bd71a7af1ecc.png)

### 6. `tabs`
There is a boxed focus so noting as part of the audit, but it seems to be the best look for the component.
![tabs focus screenshot](https://user-images.githubusercontent.com/5023024/209842839-e3cb528c-fb72-4954-b24d-39f1db4a84c1.png)

cc @driskull @macandcheese - Based on the above results, should we aim to accomplish any of the above for the January release? Thinking `link` could be a potential candidate (the others likely post 1.0).

**Comment by geospatialem** [comment]
Adding to the February priorities, but the milestone could change depending on prioritization to tackle the above components hitbox focus.

**Comment by geospatialem** [comment]
Adding one additional component per feedback from Living Atlas:

## 7. `switch`
Boxed focus, maybe could be similar to FAB's focus (radius)?
![image](https://user-images.githubusercontent.com/5023024/218182418-683a5df5-97f8-4b2b-9f75-d4b4e436b3c5.png)

**Comment by geospatialem** [comment]
@ashetland @SkyeSeitz We've gotten some feedback from a few teams regarding our focus hitbox. I performed a rough audit late last year, but would be good to get a design eye on this one, too. Could you both take a peek, and determine if the above and any other components we could the focus states on in the next few weeks?

**Comment by ashetland** [comment]
Specific to Link in this list, issue https://github.com/Esri/calcite-components/issues/6588 mentions that links used to have a +2px offset on the focus outline instead of -2px.

**Comment by ashetland** [comment]
Also related: https://github.com/Esri/calcite-components/issues/3392

**Comment by ashetland** [comment]
Recommendations in Figma: https://www.figma.com/file/VWnXYHCyCM8coZCYsCxAAt/Focus-outline-audit-%5Bissue-4633%5D?node-id=0%3A1&t=UryXS4JodeKj9btE-1

**Comment by geospatialem** [comment]
Reassigned to the July priorities to adopt all of the focus changes across the identified components in https://github.com/Esri/calcite-components/issues/4633#issuecomment-1511590646.

**Comment by geospatialem** [comment]
Most of the above have been tackled, with `rating` being prioritized for the August release.

**Comment by geospatialem** [comment]
The final component, `rating` has been mitigated on click similar to keyboard use, which has been verified on the `main` branch.
